src/storage/src/source/kafka.rs

```diff
@@ -1549,23 +1549,68 @@ mod tests {
 }
 
 /// Fetches the list of partitions and their corresponding high watermark
-fn fetch_partition_info<C: ClientContext>(
-    client: &Client<C>,
+fn fetch_partition_info<C: ConsumerContext>(
+    consumer: &BaseConsumer<C>,
```

Comment:
Ha! Using `offsets_for_time` here is clever. I didn't realize that under the hood fetching the offset for the "time" `Offset::End` is exactly equivalent to fetching the high water mark.

Reply:
I figured that out by implementing my own version of `rd_kafka_query_watermark_offsets` that can take multiple partitions and then realizing that I'm just reimplementing `rd_kafka_offsets_for_time` 😅